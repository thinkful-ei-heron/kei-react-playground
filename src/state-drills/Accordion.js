import React from 'react';
import './Accordion.css';

class Accordion extends React.Component {
  static defaultProps = {
    sections: []
  }
  state = {
    currentOpenTab: ''
  }  

  handleButtonClick(index) {
    this.setState( {currentOpenTab: index} )
  }

  renderLi = () => {
    return this.props.sections.map( (element, index) => (
      <li key={index}>
        <button onClick={() => this.handleButtonClick(index)}>
          {element.title}
        </button>
        {index===this.state.currentOpenTab && this.renderContent()}
      </li>
      )
    )
  }

  renderContent = () => {
    const currentTab = this.props.sections[this.state.currentOpenTab]
    return (
      <p className='content'>
        {currentTab.content}
      </p>
    )
  }

  render () {
    return (
      <ul>
        {this.renderLi()}
      </ul>
    )
  }
}

export default Accordion;