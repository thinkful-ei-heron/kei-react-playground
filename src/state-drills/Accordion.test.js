import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion'

describe(`Accordion Component`, () => {
  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ];
  
  it('renders empty given no sections', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders buttons only given sections', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens any clicked section and closes all others', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens any clicked section and closes all others, double', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})