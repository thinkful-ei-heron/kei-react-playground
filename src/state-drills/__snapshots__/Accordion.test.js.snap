// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component opens any clicked section and closes all others 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p
      className="content"
    >
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component opens any clicked section and closes all others, double 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p
      className="content"
    >
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion Component renders buttons only given sections 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no sections 1`] = `<ul />`;
